Tests/Tests/STPSourceParamsTest.m

```diff
@@ -46,4 +48,29 @@ - (void)testCardParamsWithCard {
     XCTAssertEqualObjects(sourceAddress[@"country"], card.addressCountry);
 }
 
+- (NSString *)redirectMerchantNameQueryItemValueFromURLString:(NSString *)urlString {
+    NSURLComponents *components = [NSURLComponents componentsWithString:urlString];
+    for (NSURLQueryItem *item in components.queryItems) {
+        if ([item.name isEqualToString:@"redirect_merchant_name"]) {
+            return item.value;
+        }
+    }
+    return nil;
+}
+
+- (void)testRedirectMerchantNameURL {
+    STPSourceParams *sourceParams = [STPSourceParams sofortParamsWithAmount:1000
+                                                                  returnURL:@"test://foo?value=baz"
+                                                                    country:@"DE"
+                                                        statementDescriptor:nil];
+
+    NSDictionary *params = [STPFormEncoder dictionaryForObject:sourceParams];
+    // Should be nil because we have no app name in tests
+    XCTAssertNil([self redirectMerchantNameQueryItemValueFromURLString:params[@"redirect"][@"return_url"]]);
+
+    sourceParams.redirectMerchantName = @"bar";
+    params = [STPFormEncoder dictionaryForObject:sourceParams];
+    XCTAssertEqualObjects([self redirectMerchantNameQueryItemValueFromURLString:params[@"redirect"][@"return_url"]], @"bar");
+}
```

Comment:
could we also add a test that makes sure we don't overwrite an existing `redirect_merchant_name` parameter?